lib/Doctrine/ORM/UnitOfWork.php

```diff
@@ -3348,7 +3352,7 @@ private function mergeEntityStateIntoManagedCopy($entity, $managedCopy)
     {
         $class = $this->em->getClassMetadata(get_class($entity));
 
-        foreach ($class->reflClass->getProperties() as $prop) {
+        foreach ($class->getAllReflectionProperties() as $prop) {
```

Comment:
This may be replaced with a static utility if the classmetadata changes are not required